### PR TITLE
Add BC, UO and misc doco to THM

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #19737

### CODE REVIEW

File: modules/thermal_hydraulics/doc/content/source/base/Simulation.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Simulation
+
+Simulation is one of the two base classes of a `THMProblem`. It implements a number of the specificities
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/Simulation.md:3]:
```suggestion
Simulation is one of the two base classes of a [THMProblem.md]. It implements a number of the specificities
```

File: modules/thermal_hydraulics/doc/content/source/base/Simulation.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+of a thermal hydraulics solve, notably the reliance on `Components` to form the problem.
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/Simulation.md:4]:
```suggestion
of a thermal hydraulics solve, notably the reliance on [Components](syntax/Components/index.md) to form the problem.
```

File: modules/thermal_hydraulics/doc/content/source/base/Simulation.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+It is in charge of :
+
+- building the mesh, created by each [Component](syntax/Components/index.md)
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/Simulation.md:8]:
```suggestion
- building the mesh, each part being created by each [Component](syntax/Components/index.md)
```

File: modules/thermal_hydraulics/doc/content/source/base/Simulation.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- adding the variables
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/Simulation.md:9]:
```suggestion
- creating the variables
```

File: modules/thermal_hydraulics/doc/content/source/base/Simulation.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- setting initial conditions for the variables, whether from file or from user input
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/Simulation.md:10]:
```suggestion
- setting up initial conditions for the variables, whether from file or from user input
```

File: modules/thermal_hydraulics/doc/content/source/base/Simulation.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+- add components
+- initialize components
+- add MOOSE objects created by components
+- identify component loops
+- set up the coordinate system
+
+
+It also performs integrity checks for:
+
+- the coupling matrix, making sure it's sufficiently capture inter-variable dependencies
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/Simulation.md:24]:
```suggestion
- the coupling matrix, making sure it sufficiently captures inter-variable dependencies
```

File: modules/thermal_hydraulics/doc/content/source/bcs/ADHeatFlux3EqnBC.md [hunk 7]
```diff
@@ -1,20 +1,18 @@
...
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /BCs/ADHeatFlux3EqnBC
 
-## Overview
-
-!! Replace these lines with information regarding the ADHeatFlux3EqnBC object.
+The heat flux contribution to the residual $R_i$ for the weak form is computed as:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/ADHeatFlux3EqnBC.md:5]:
```suggestion
The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:
```

File: modules/thermal_hydraulics/doc/content/source/bcs/ADHSHeatFluxBC.md [hunk 8]
```diff
@@ -1,20 +1,19 @@
...
-!alert construction title=Undocumented Class
-The ADHSHeatFluxBC has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /BCs/ADHSHeatFluxBC
 
-## Overview
+The heat flux contribution to the residual $R_i$ for the weak form is computed as:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/ADHSHeatFluxBC.md:5]:
```suggestion
The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:
```

File: modules/thermal_hydraulics/doc/content/source/bcs/ADConvectionHeatTransferBC.md [hunk 9]
```diff
@@ -1,20 +1,21 @@
...
-!alert construction title=Undocumented Class
-The ADConvectionHeatTransferBC has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /BCs/ADConvectionHeatTransferBC
 
-## Overview
+The heat flux contribution to the residual $R_i$ for the weak form is computed as:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/ADConvectionHeatTransferBC.md:5]:
```suggestion
The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:
```

File: modules/thermal_hydraulics/doc/content/source/bcs/ADBoundaryFlux3EqnBC.md [hunk 10]
```diff
@@ -1,20 +1,19 @@
...
-The ADBoundaryFlux3EqnBC has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /BCs/ADBoundaryFlux3EqnBC
 
-## Overview
+The boundary conditions retrieves the flux from an `ADBoundaryFluxBase`-derived user object,
+such as the [ADBoundaryFlux3EqnFreeOutflow.md].
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/ADBoundaryFlux3EqnBC.md:7]:
```suggestion

The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:

```

File: modules/thermal_hydraulics/doc/content/source/bcs/ConvectionHeatTransferBC.md [hunk 11]
```diff
@@ -1,20 +1,20 @@
...
-!alert construction title=Undocumented Class
-The ConvectionHeatTransferBC has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /BCs/ConvectionHeatTransferBC
 
-## Overview
+The heat flux contribution to the residual $R_i$ for the weak form is computed as:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/ConvectionHeatTransferBC.md:5]:
```suggestion
The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:
```

File: modules/thermal_hydraulics/doc/content/source/bcs/HeatStructure2DRadiationCouplerRZBC.md [hunk 12]
```diff
@@ -1,20 +1,37 @@
...
-!alert construction title=Undocumented Class
-The HeatStructure2DRadiationCouplerRZBC has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /BCs/HeatStructure2DRadiationCouplerRZBC
 
-## Overview
+The heat flux contribution to the residual $R_i$ for the weak form is computed as:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/HeatStructure2DRadiationCouplerRZBC.md:5]:
```suggestion
The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:
```

File: modules/thermal_hydraulics/doc/content/source/bcs/HeatStructure2DRadiationCouplerRZBC.md [hunk 13]
```diff
@@ -1,20 +1,37 @@
...
+
+where $\psi_i$ are the test functions, $\sigma$ is the Stefan-Boltzmann constant, $T$ is one heat structure temperature
+variable, $T_{coupled}$ the other heat structure temperature variable, $R$ the radiation resistance, and $C$ the
+circumference of the heat structure boundary considered.
+
+The radiation resistance is computed as:
+
+!equation
+R = \dfrac{1.0 - \epsilon}{\epsilon} + \dfrac{1.0}{V} +
+    \dfrac{1.0 - \epsilon_{coupled}} / \epsilon_{coupled} * A / A_{coupled}
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/bcs/HeatStructure2DRadiationCouplerRZBC.md:19]:
```suggestion
    \dfrac{1.0 - \epsilon_{coupled}}{\epsilon_{coupled}}  \dfrac{A}{A_{coupled}}
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedCompressor1PhaseUserObject.md [hunk 14]
```diff
@@ -1,20 +1,49 @@
...
-documentation clear for users.
-
 !syntax description /UserObjects/ADShaftConnectedCompressor1PhaseUserObject
 
-## Overview
+A compressor is a type of volume junction, a [ADVolumeJunction1PhaseUserObject.md].
+As such its base contribution to the residual and Jacobian is set by this class. An additional contribution to
+the residual (and Jacobian) is described below:
+
+The contribution to the residual of the energy equation is simply the power dissipated in the component:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedCompressor1PhaseUserObject.md:9]:
```suggestion
The additional contribution to the residual of the energy equation is simply the power dissipated in the component:
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedCompressor1PhaseUserObject.md [hunk 15]
```diff
@@ -1,20 +1,49 @@
...
+
+!equation
+R_e += (T_{friction} + T_{dissipation}) \omega
+
+where $T_{friction}$ is the friction torque, $T_{dissipation}$ is the dissipation torque and $\omega$ is the rotation speed of
+the compressor.
+
+The contribution to the residual of the momentum equation in each direction is:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedCompressor1PhaseUserObject.md:17]:
```suggestion
The additional contribution to the residual of the momentum equation in each direction is:
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md [hunk 16]
```diff
@@ -1,20 +1,45 @@
...
-
 !syntax description /UserObjects/ADShaftConnectedPump1PhaseUserObject
 
-## Overview
+A pump is a type of volume junction, a [ADVolumeJunction1PhaseUserObject.md].
+As such its base contribution to the residual and Jacobian is set by this class. An additional contribution to
+the residual (and Jacobian) is described below:
+
+
+The contribution to the residual of the energy equation is simply the power of the pump:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md:10]:
```suggestion
The additional contribution to the residual of the energy equation is simply the power of the pump:
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md [hunk 17]
```diff
@@ -1,20 +1,45 @@
...
+
+!equation
+R_e += (T_{hydraulic} + T_{dissipation}) \omega
+
+where $T_{hydraulic}$ is the friction torque, and $\omega$ is the rotation speed of
+the pump.
+
+The contribution to the residual of the momentum equation in each direction is:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md:18]:
```suggestion
The additional contribution to the residual of the momentum equation is:
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md [hunk 18]
```diff
@@ -1,20 +1,45 @@
...
+
+!equation
+\vec{R} = \dfrac{\rho A}{A} \vec{g} H A_{ref} \vec{d}_{out}
+
+where $\rho A$ is the conserved density, $A$ the local flow area , $A_{ref}$ is the reference flow area, $\vec{g}$ is the gravity vector,
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md:23]:
```suggestion
where $\rho A$ is the conserved density, $A$ the local flow area, $A_{ref}$ is the reference flow area, $\vec{g}$ is the gravity vector,
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md [hunk 19]
```diff
@@ -1,20 +1,45 @@
...
+$H$ the pupmp head, and $\vec{d}_{out}$ the outlet direction.
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md:24]:
```suggestion
$H$ the pump head, and $\vec{d}_{out}$ the outlet direction.
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md [hunk 20]
```diff
@@ -1,20 +1,45 @@
...
+
+The pump head is computed as:
+
+!equation
+H =  (\dfrac{\omega}{\omega_{rated}})^2 + (\dfrac{\dfrac{\rho u A A}{A}}{Q_{rated}})^2 ) H H_{rated}
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md:29]:
```suggestion
H_{pump} =  ( (\dfrac{\omega}{\omega_{rated}})^2 + (\dfrac{\dfrac{\rho u A A}{\rho A}}{Q_{rated}})^2 ) H_{function} H_{rated}
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md [hunk 21]
```diff
@@ -1,20 +1,45 @@
...
+
+where $\omega_{rated}$ is the rated rotation speed of the pump, $\rho$ the fluid density, $u$ the fluid velocity, $A$ the local area,
+$Q_{rated}$ is the rated flow rate and $H_{rated}$ is the rated head of the pump.
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedPump1PhaseUserObject.md:32]:
```suggestion
$Q_{rated}$ is the rated flow rate, $H_{function}$ an adimensional pump head function and $H_{rated}$ is the rated head of the pump.
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedTurbine1PhaseUserObject.md [hunk 22]
```diff
@@ -1,20 +1,46 @@
...
-!alert construction title=Undocumented Class
-The ADShaftConnectedTurbine1PhaseUserObject has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /UserObjects/ADShaftConnectedTurbine1PhaseUserObject
 
-## Overview
+A turbine is a parallel channel junction, a [ADJunctionParallelChannels1PhaseUserObject.md], a type of volume junction,
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedTurbine1PhaseUserObject.md:5]:
```suggestion
A turbine is a volume junction, derived from 
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedTurbine1PhaseUserObject.md [hunk 23]
```diff
@@ -1,20 +1,46 @@
...
+a [ADVolumeJunction1PhaseUserObject.md]. As such its base contribution to the residual and Jacobian is set by these classes.
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADShaftConnectedTurbine1PhaseUserObject.md:6]:
```suggestion
[ADVolumeJunction1PhaseUserObject.md]. As such its base contribution to the residual and Jacobian is set by these classes.
```

File: modules/thermal_hydraulics/src/userobjects/ADShaftConnectedPump1PhaseUserObject.C [hunk 24]
```diff
@@ -186,7 +186,7 @@ ADShaftConnectedPump1PhaseUserObject::computeFluxesAndResiduals(const unsigned i
 
     // a positive head value results in a positive S_momentum
     const ADRealVectorValue S_momentum = (_rhoA[0] / _A[0]) * _g * _pump_head * _A_ref * _di_out;
-    //
+    // SEEMS WRONG
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/userobjects/ADShaftConnectedPump1PhaseUserObject.C:189]: not sure I remember why

Comment (joshuahansel) [reply modules/thermal_hydraulics/src/userobjects/ADShaftConnectedPump1PhaseUserObject.C:189]: Probably should remove if you can't remember why.

File: modules/thermal_hydraulics/src/userobjects/ADNumericalFlux3EqnBase.C [hunk 25]
```diff
@@ -13,6 +13,9 @@ InputParameters
 ADNumericalFlux3EqnBase::validParams()
 {
   InputParameters params = GeneralUserObject::validParams();
+  params.addClassDescription(
+      "Base class for retrieving numerical fluxes in the the three-equation model");
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/src/userobjects/ADNumericalFlux3EqnBase.C:17]: Does it do anything to add class descriptions to un-registered classes like base classes?

File: modules/thermal_hydraulics/doc/content/source/problems/THMProblem.md [hunk 26]
```diff
@@ -1,20 +1,9 @@
...
 
-## Overview
-
-!! Replace these lines with information regarding the THMProblem object.
-
-## Example Input File Syntax
-
-!! Describe and include an example of how to use the THMProblem object.
+The `THMProblem` is composed of a regular [FEProblem.md] and a [Simulation.md] class.
+When the `THMProblem` advances state, the advances the `FEProblem` then the `Simulation` state.
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/problems/THMProblem.md:6]:
```suggestion
When the `THMProblem` advances state, this advances the `FEProblem` and then the `Simulation` state.
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADHeatFluxFromHeatStructure3EqnUserObject.md [hunk 27]
```diff
@@ -1,20 +1,18 @@
...
-!alert construction title=Undocumented Class
-The ADHeatFluxFromHeatStructure3EqnUserObject has not been documented. The content listed below should be used as a starting point for
-documenting the class, which includes the typical automatic documentation associated with a
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /UserObjects/ADHeatFluxFromHeatStructure3EqnUserObject
 
-## Overview
+The heat flux is computed as:
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADHeatFluxFromHeatStructure3EqnUserObject.md:5]:
```suggestion
The heat flux $q$ is computed as:
```

File: modules/thermal_hydraulics/doc/content/source/userobjects/ADHeatFluxFromHeatStructure3EqnUserObject.md [hunk 28]
```diff
@@ -1,20 +1,18 @@
...
 
-!! Replace these lines with information regarding the ADHeatFluxFromHeatStructure3EqnUserObject object.
+!equation
+H_{wall} (T_{wall} - T)
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/userobjects/ADHeatFluxFromHeatStructure3EqnUserObject.md:8]:
```suggestion
q = H_{wall} (T_{wall} - T)
```

File: modules/thermal_hydraulics/doc/content/source/kernels/ADOneDEnergyWallHeatFlux.md [hunk 29]
```diff
@@ -7,7 +7,8 @@ The heat flux contribution to the residual $R_i$ for the weak form is computed a
 \begin{equation}
 R_i = (\psi_i, -q_{wall} P_{hf}) \quad \forall \psi_i,
 \end{equation}
-where $\psi_i$ are the test functions and $q_{wall}$ is the local heat flux, and $P_{hf}$ the
+
+where $\psi_i$ are the test functions, $q_{wall}$ is the local heat flux, and $P_{hf}$ the
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/kernels/ADOneDEnergyWallHeatFlux.md:11]:
```suggestion
where $\psi_i$ are the test functions, $q_{wall}$ is the local heat flux, and $P_{hf}$ is the
```

File: modules/thermal_hydraulics/doc/content/source/bcs/ADHSHeatFluxBC.md [hunk 30]
```diff
@@ -1,20 +1,19 @@
...
-documentation clear for users.
-
 !syntax description /BCs/ADHSHeatFluxBC
 
-## Overview
+The heat flux contribution to the residual $R_i$ for the weak form of the energy equation is computed as:
 
-!! Replace these lines with information regarding the ADHSHeatFluxBC object.
+\begin{equation}
+R_i = (\psi_i, - S_{pp} * S_{fn}(t, \vec{x}) F(t, \vec{x})) \quad \forall \psi_i,
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/bcs/ADHSHeatFluxBC.md:8]:
```suggestion
R_i = (\psi_i, - S_{pp} S_{fn}(t, \vec{x}) F(t, \vec{x})) \quad \forall \psi_i,
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: modules/thermal_hydraulics/src/userobjects/ADNumericalFlux3EqnBase.C (hunk 25 above)
Comment (GiudGiud) [reply modules/thermal_hydraulics/src/userobjects/ADNumericalFlux3EqnBase.C:17]: not really.

Comment (joshuahansel) [reply modules/thermal_hydraulics/src/userobjects/ADNumericalFlux3EqnBase.C:17]: Doesn't hurt anything, so I'm not asking you to remove, but I was just curious.

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**